src/layers/linear.rs

```diff
@@ -26,9 +31,47 @@ impl Layer<Ix1, Ix1> for Linear {
         Tensor::from_vec(params)
     }
 
+    fn weights(&self) -> Vec<f64> {
+        self.weights.iter().map(|v| v.value()).collect()
+    }
+
+    fn biases(&self) -> Vec<f64> {
+        self.biases.iter().map(|v| v.value()).collect()
+    }
+
     fn forward(&self, input: &Tensor<Ix1>) -> Tensor<Ix1> {
         let weights_t = self.weights.t().into_owned();
 
         input.dot(&weights_t) + &self.biases
     }
+
+    fn set_weights(&mut self, new_weights: Vec<f64>) {
+        if self.weights.len() != new_weights.len() {
+            panic!("Loaded weight count does not match layer weight count.");
+        }
+
+        self.weights
+            .iter_mut()
+            .zip(new_weights.iter())
+            .for_each(|(value_num, num)| *value_num = val!(*num));
```

Comment:
To not have to dereference `num` you can do `zip(new_weights.into_iter())`. Here is my suggestion
```suggestion
        let mut iter = new_weights.into_iter();
        self.weights
            .map_inplace(|v| *v = val!(iter.next().unwrap()));
```